hnn_core/network.py

```diff
@@ -751,35 +792,34 @@ def _instantiate_drives(self, n_trials=1):
         for trial_idx in range(n_trials):
             for drive in self.external_drives.values():
                 event_times = list()  # new list for each trial and drive
-
-                # loop over drives (one for each target cell population)
-                # and create event times
-                for this_cell_drive_conn in drive['conn'].values():
-                    if drive['cell_specific']:
+                if drive['cell_specific']:
+                    # loop over drives (one for each target cell population)
+                    # and create event times
+                    for this_cell_drive_conn in drive['conn'].values():
+                        target_type = this_cell_drive_conn['target_type']
                         for drive_cell_gid in this_cell_drive_conn['src_gids']:
                             event_times.append(_drive_cell_event_times(
-                                drive['type'], this_cell_drive_conn,
-                                drive['dynamics'], trial_idx=trial_idx,
+                                drive['type'],
+                                drive['dynamics'],
+                                self.cell_response.times[-1],
```

Comment:
instead of doing this, we should raise an error after the call to `simulate_dipole` if any of the `drive['dynamics']['tstop']` exceeds `tstop`. Right now, it's really weird that `cell_response.times[-1]` is used in Network, and I don't want to tangle it even more

Reply:
The `tstop` value here is the simulation stop time, not the drive stop time (see above for the rationale). My understanding was that we would eventually do away with passing around `Network._params['tstop']`, however, I thought this would be safe since `Network.cell_response.times` will always need to be updated even if `tstop` is modified in `simulate_dipole()`. Any other suggestions?

Reply:
humm ... we also discussed [separating cell_response from Network](https://github.com/jonescompneurolab/hnn-core/issues/341#issuecomment-852304024). I'm not sure if anyone is motivated enought to implement it though. I can live with this for now

Reply:
That's fair. I'm down to work on separating cell_response from Network after 0.2.